src/main/java/ch/njol/skript/lang/SyntaxElement.java

```diff
@@ -47,4 +51,22 @@ default ParserInstance getParser() {
 		return ParserInstance.get();
 	}
 
+	/**
+	 * Gets the structures in which this element can be used in. If there are no limits to this element, you may
+	 * return an empty array, or simply not override this method.
+	 * @return the structures this element is usable in, or an empty list.
+	 */
+	default List<Class<? extends Structure>> getUsableStructures() {
```

Comment:
Could there be a way to specify the error? Something like `getUsableStructures(String defaultError)` or something.

Reply:
unfortunately i don't think so as you're taking in the parameter instead of returning it, and returning both in a pair of such would be a bit of a pain i think, but it does seem like a nice feature to have so if someone has an impl that would be nice to hear !